### PR TITLE
PM-7288: Scan workflow change in support of scanning maven projects with SonarCloud

### PR DESCRIPTION
**Pull request by bwdil**
PM-7288: Scan workflow change in support of scanning maven projects with SonarCloud

- Include SonarCloud properties for Maven project
- Inclusion of `sonar-maven-plugin` in `pom.xml`
- Required changes to actions in `scan.yml`

### CODE REVIEW

File: .github/workflows/scan.yml [hunk 1]
```diff
@@ -60,19 +60,20 @@ jobs:
...
         with:
           fetch-depth: 0
           ref: ${{  github.event.pull_request.head.sha }}
-
-      - name: Scan with SonarCloud
-        uses: sonarsource/sonarcloud-github-action@49e6cd3b187936a73b8280d59ffd9da69df63ec9 # v2.1.1
-        env:
-          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up Java
```
Comment (withinfocus) [inline .github/workflows/scan.yml:67]: ⛏️ Please add newlines between steps.

File: .github/workflows/scan.yml [hunk 2]
```diff
@@ -60,19 +60,20 @@ jobs:
...
+        uses: actions/setup-java@99b8673ff64fbf99d8d325f52d9a5bdedb8483e9 # v4.2.1
         with:
-          args: >
-            -Dsonar.organization=${{ github.repository_owner }}
-            -Dsonar.projectKey=${{ github.repository_owner }}_${{ github.event.repository.name }}
-            -Dsonar.sources=src/,examples/
-            -Dsonar.tests=src/test/
+          java-version: 17
+          distribution: "temurin"
+      - name: Build and Analyze
```
Comment (withinfocus) [inline .github/workflows/scan.yml:72]:
```suggestion
      - name: Scan with SonarCloud
```

File: pom.xml [hunk 3]
```diff
@@ -216,6 +219,11 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.sonarsource.scanner.maven</groupId>
```
Comment (withinfocus) [inline pom.xml:223]: ⛏️ Needs formatting.

File: pom.xml [hunk 4]
```diff
@@ -45,6 +45,9 @@
     <properties>
         <java.version>1.8</java.version>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <sonar.organization>bitwarden</sonar.organization>
+        <sonar.host.url>https://sonarcloud.io</sonar.host.url>
+        <sonar.projectKey>bitwarden_passwordless-java</sonar.projectKey>
```
Comment (withinfocus) [inline pom.xml:50]: ❓ With these above in the Sonar invocation can they be removed here?

**Review by withinfocus — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 4 above)
Comment (bwdil) [reply pom.xml:50]: These are required in the pom.xml by SonarCloud to support maven projects.

File: pom.xml (hunk 3 above)
Comment (bwdil) [reply pom.xml:223]: The format should be correct. Is something wrong with it?

File: .github/workflows/scan.yml [hunk 5]
```diff
@@ -60,19 +60,21 @@ jobs:
 
     steps:
       - name: Check out repo
-        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
         with:
           fetch-depth: 0
           ref: ${{  github.event.pull_request.head.sha }}
+      - name: Set up Java
```
Comment (withinfocus) [inline .github/workflows/scan.yml:67]: ⛏️ Still need a newline here.

File: pom.xml (hunk 3 above)
Comment (withinfocus) [reply pom.xml:223]: Needs leading whitespace.

**Review by withinfocus — CHANGES_REQUESTED** [review]

File: .github/workflows/scan.yml (hunk 5 above)
Comment (bwdil) [reply .github/workflows/scan.yml:67]: oh weird, I thought I saved my changes.

**Review by withinfocus — APPROVED** [review]

**Pull request merged**